### ISSUE TITLE
Large title bar on macOS Sonoma when using railwaycat/homebrew-emacsmacport and Doom Emacs

### ISSUE THREAD

**Issue by adamliter** [doomemacs/doomemacs#7532]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

The expected behavior is to have a normal-sized title bar when opening Doom Emacs on macOS Sonoma. After upgrading to macOS Sonoma, something has changed, and it seems like a bug in Doom Emacs, based on [the discussion here](https://discourse.doomemacs.org/t/how-to-have-tool-bar-mode-0-apply-at-startup-to-avoid-large-title-bar-on-macos-sonoma-when-using-railwaycat-homebrew-emacsmacport/4222).

This is also discussed in https://github.com/railwaycat/homebrew-emacsmacport/issues/362.

### Current behavior

Opening Doom Emacs on macOS Sonoma leads to a title bar of this size:

![image](https://user-images.githubusercontent.com/41767537/270948850-7d9ea814-f1a3-408b-a4cf-afb21c41c072.png)

However, running `/Applications/Emacs.app/Contents/MacOS/Emacs -q --eval '(tool-bar-mode 0)' &` leads to a title bar of this size:

![image](https://user-images.githubusercontent.com/4974404/277714260-0b7841a2-437a-421c-a90a-7f14000cdb22.png)

### Steps to reproduce

1. `brew install emacs-mac --with-natural-title-bar`
2. `/Applications/Emacs.app/Contents/MacOS/Emacs -q` to get large title bar
3. `/Applications/Emacs.app/Contents/MacOS/Emacs -q --eval '(tool-bar-mode 0)'` to get small/reasonable title bar
4. `/Applications/Emacs.app/Contents/MacOS/Emacs` to get large title bar with Doom configuration (regardless of whether config.el contains `(tool-bar-mode 0)`)

Adding `(add-hook 'doom-after-init-hook (lambda () (tool-bar-mode 1) (tool-bar-mode 0)))` to `config.el` is a workaround as suggested by Dolerich in the linked Discourse post.

### System Information

https://pastebin.com/embed_js/ZGGDx9NA

**Comment by qcfu-bu** [comment]
I don't think this is a doom issue and is more of a macos sonoma issue. I have this problem on my own custom config as well.

**Comment by DivineDominion** [comment]
In case someone else googles this:

The underlying issue is that the toolbar needs to be disabled before the window title bar style is changed. If you don't have the patched Emacs binary, you can toggle `M-x tool-bar-mode` on and off again interactively. That will be long after the startup sequence has set the window title and shrink the title bar.

**Comment by caseycronyn** [comment]
This is still an issue for me. Here is the code that fixes it if anyone needs, from [here](https://discourse.doomemacs.org/t/how-to-have-tool-bar-mode-0-apply-at-startup-to-avoid-large-title-bar-on-macos-sonoma-when-using-railwaycat-homebrew-emacsmacport/4222/2) `(add-hook 'doom-after-init-hook (lambda () (tool-bar-mode 1) (tool-bar-mode 0)))`

**Comment by DamianB-BitFlipper** [comment]
1) Make sure you installed emacs via `brew` and the `emacs-mac` source.
2) Update your emacs installation.

Using latest `emacs-mac` works for me. I have no oversized tool bar with no extra configuration needed.
